app/experimenter/docs/adrs/0002-use-react-bootstrap-cra-gql-for-nimbus.md

```diff
@@ -0,0 +1,113 @@
+# Use React, Bootstrap, CRA, and GQL for the Nimbus Front-End Stack
+
+* Status: proposed
+* Deciders: Lauren Zugai, Jody Heavener, [TBD]
+* Date: 2020-09-08
+
+## Context and Problem Statement
+
+The front-end for Experimenter is currently composed of two applications. A more mature application for Normandy experiments, found at `app/experimenter/static/core`, is composed of a mix of Django templates, vanilla JS, jQuery, and React components. A newer React application for Nimbus experiments (previously named Rapid experiments) can be found at `app/experimenter/static/rapid`.
+
+With the decision made to build the Nimbus console as part of the Experimenter codebase ([see ADR #0001](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md)), it was determined we likely want a "new front-end." This ADR, while slightly atypical as it does not serve to document a single architectural change, outlines a proposed directory tree change regarding the front-end, a proposed new front-end stack, how it compares to the current `rapid` stack, and how we will benefit from these changes.
+
+## Decision Drivers
+
+Drivers for directory tree changes:
+* A more obvious separation of concerns for:
+  * the front-end and back-end
+  * legacy front-end code and Nimbus front-end code
+
+Drivers for the proposed new stack:
+* Better development tooling
+* A more cohesive fit into the FxA team's stack and processes
+* Development speed
+
+Drivers for starting a new application:
+* Time needed and ease of refactoring the current `rapid` React application vs time needed to start with a new application
+
+## Proposed Changes
+
+### App Directory Tree
+
+Currently, a `static/` directory contains the front-end applications with some shared assets, and a `templates/` directory lives adjacent to it.  To provide a more obvious separation of concerns in the directory tree:
+
+* Rename `static/` to `legacy-ui/` since `static/rapid/` will become obsolete and `static/core/` will be the legacy Normandy platform
+* Move `templates/` into `legacy-ui/`
```

Comment:
I might suggest throwing a README inside `legacy-ui` as well to document what that directory is, why it's there, etc. Maybe this ADR suffices, I just know there's a lot happening in `app/experimenter/`, so any clarification is nice to have.

Reply:
👍 Worth doing. We should also keep a README in `nimbus-ui` up to date and easy to work with like Settings IMO.

Reply:
For someone that just got started with the project, I honestly was a little confused by (well firstly where to find the front-end, ha) `core` and `rapid`. I think if we stuck with those that could be mitigated by a quick explanation in the root README, but:

>  I'd suggest we also rename any other references in the code that currently use 'core' and 'rapid',

I'd be fine with us doing this personally. I was going to suggest that a happy medium could potentially be changing `rapid` to `nimbus` and leaving `core` as-is, but as "nimbus" is to be the shiny new centralized location for things, maybe "core" would be confusing later once Nimbus is built out?

Reply:
Yeah and `core` gets extra confusing now that we're trying to split up the models and in theory you have something like `ExperimentBase` as the base model and `ExperimentCore` `ExperimentRapid` as the children models, `core` and `base` are ambiguous 😆 

Whoever said that naming was the hardest problem in software 🧐

Reply:
I was trying to avoid the name nimbus in the code becuase it's the code name, but it could also be just as confusing to see 'rapid' everywehre when we know its' called 'nimbus' so yeah maybe renaming everything to 'Legacy' and 'Nimbus' is the right call.  I'll have to `awk/sed` this whole model pr, or maybe we can do that as a separate pr.....

Reply:
Ah, that makes sense on the "nimbus" naming hesitation since it's a code name. That didn't even cross my mind honestly since I've only ever seen it called that. 😅

I do personally like "nimbus" and "legacy" but I'm still open to other suggestions. 👍

Reply:
Looking at this model refactor PR more, I don't think it'll be feasible (or imminently practical) to rename the entire Python code base, but I also agree that 'core' and 'rapid' are confusing and we don't necessarily need to be locked into them.  I'd say disregard what I said earlier about maintaining consistency and use the names you prefer 👍